xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/config/XmRestTemplateConfiguration.java

```diff
@@ -25,10 +25,28 @@ public class XmRestTemplateConfiguration {
 
     public static final String XM_CONFIG_REST_TEMPLATE = "xm-config-rest-template";
 
-    @Bean(XM_CONFIG_REST_TEMPLATE)
-    public RestTemplate restTemplate(RestTemplateCustomizer customizer) {
+    @ConditionalOnProperty(value = "spring.cloud.loadbalancer.enabled", havingValue = "true", matchIfMissing = true)
+    static class XmLoadBalancerRestTemplateConfiguration {
+
+        @Bean(XM_CONFIG_REST_TEMPLATE)
+        public RestTemplate restTemplate(RestTemplateCustomizer customizer) {
+            RestTemplate restTemplate = createRestTemplate();
+            customizer.customize(restTemplate);
+            return restTemplate;
+        }
+    }
+
+    @ConditionalOnProperty(value = "spring.cloud.loadbalancer.enabled", havingValue = "false")
+    static class XmPlainRestTemplateConfiguration {
+
+        @Bean(XM_CONFIG_REST_TEMPLATE)
+        public RestTemplate restTemplate() {
```

Comment:
RestTemplateCustomizer is a powerful and useful tool to tune RestTempalte. FE Spring Cloud Sleuth uses it for http enhancement

Reply:
Yes, you are right - sleuth uses customizer, but not loadbalancer customizer. Sleuth use `org.springframework.boot.web.client.RestTemplateCustomizer` and here we have `org.springframework.cloud.client.loadbalancer.RestTemplateCustomizer`